test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,249 @@
+                        prefix_key += random_generator::instance().generate_string(tc->key_size);
+                        if (tc->insert(cursor, coll.id, prefix_key)) {
+                            /* We failed to insert, rollback our transaction and retry. */
+                            tc->transaction.rollback();
+                            --count;
+                            continue;
+                        }
+                        /* Commit txn at commit timestamp 100. */
+                        tc->transaction.commit("commit_timestamp=" + tc->tsm->decimal_to_hex(100));
+                    }
+                }
+            }
+        }
+    }
+
+    public:
+    search_near_01(const test_harness::test_args &args) : test(args) {}
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *tsm,
+      test_harness::configuration *config, test_harness::workload_tracking *tracking) override final
+    {
+        int64_t collection_count, key_count, key_size;
+        std::vector<thread_context *> workers;
+        thread_manager tm;
+
+        /* Validate our config. */
+        collection_count = config->get_int(COLLECTION_COUNT);
+        key_count = keys_per_prefix = config->get_int(KEY_COUNT_PER_COLLECTION);
+        key_size = config->get_int(KEY_SIZE);
+        testutil_assert(key_size > 0);
+        /* Keys must be unique. */
+        testutil_assert(key_count <= pow(10, key_size));
+
+        logger::log_msg(LOG_INFO,
+          "Populate configuration with key size: " + std::to_string(key_size) +
+            " key count: " + std::to_string(key_count) +
+            " number of collections: " + std::to_string(collection_count));
+
+        /* Create n collections as per the configuration. */
+        for (int64_t i = 0; i < collection_count; ++i)
+            /*
+             * The database model will call into the API and create the collection, with its own
+             * session.
+             */
+            database.add_collection();
+
+        /*
+         * Spawn thread_count threads to populate the database, theoretically we should be IO bound
+         * here.
+         */
+        for (int64_t i = 0; i < ALPHABET_SIZE; ++i) {
+            thread_context *tc = new thread_context(i, thread_type::INSERT, config,
+              connection_manager::instance().create_session(), tsm, tracking, database);
+            workers.push_back(tc);
+            tm.add_thread(
+              std::bind(&search_near_01::populate_worker, this, std::placeholders::_1), tc);
+        }
+
+        /* Wait for our populate threads to finish and then join them. */
+        logger::log_msg(LOG_INFO, "Populate: waiting for threads to complete.");
+        tm.join();
+
+        /* Cleanup our workers. */
+        for (auto &it : workers) {
+            delete it;
+            it = nullptr;
+        }
+
+        /* Force evict all the populated keys in all of the collections. */
```

Comment:
This is the issue we were discussing today. Can you comment that code and see if your test still passes? We should leave eviction to be handled automatically, what do you think ?

Reply:
Just tried turning eviction off, and the prefix fast path stat doesn't get incremented. I think this is because it will follow the path to find values on the update list, which doesn't increment the prefix fast path. So I think this doesn't work as of now, until we add the early exit when we are parsing the update list.

Reply:
Are you talking about WT-8092 ?

Reply:
Yeah, to remove the eviction portion of the test we will need to complete WT-8106 as well.